dataline-workers/src/main/java/io/dataline/workers/protocols/singer/SingerTap.java

```diff
@@ -22,15 +22,24 @@
  * SOFTWARE.
  */
 
-package io.dataline.workers;
+package io.dataline.workers.protocols.singer;
 
 import io.dataline.config.StandardTapConfig;
+import io.dataline.singer.SingerMessage;
+import io.dataline.workers.InvalidCredentialsException;
+import java.io.IOException;
 import java.nio.file.Path;
-import java.util.stream.Stream;
+import java.util.Optional;
 
-public interface TapFactory<T> {
+public interface SingerTap extends AutoCloseable {
```

Comment:
I think I preferred SingerProtocol tap just to avoid confusion. but nbd if it's too verbose.